components/apimgt/org.wso2.carbon.apimgt.throttling.siddhi.extension/src/test/java/org/wso2/carbon/apimgt/throttling/siddhi/extension/CountAttributeAggregatorWithResetTestCase.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.throttling.siddhi.extension;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.wso2.siddhi.core.ExecutionPlanRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.query.output.callback.QueryCallback;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.util.EventPrinter;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CountAttributeAggregatorWithResetTestCase {
+    private static final Log log = LogFactory.getLog(CountAttributeAggregatorWithResetTestCase.class);
+    private AtomicInteger atomicEventCount;
+
+    @Before
+    public void init() {
+        atomicEventCount = new AtomicInteger(0);
+    }
+
```

Comment:
Java doc comments missing. Mention what is tested by this test

Reply:
Do for other test methods as well

Reply:
Added meaningful method names with https://github.com/wso2/carbon-apimgt/pull/12443/commits/a583c6ae8e5be9a7a7879183801ad442b86123e4#diff-a92a928154247caabe1158c2ac06adfeab5798b6f9e50538cb203bffabb50a70